woocommerce/wc-admin
# Fixing local WCPay payment task and refactoring for dumber components

Addressing part of #7121

This involves a minor refactor of some of the logic for remote payments. This was done partially to reduce the need for `select` logic in various downstream components, and reduce data and props being passed around. 

This also allowed me to address some issues that had cropped up in `WCPayMethodCard.js` for local payments, since it was utilizing some logic specific to remote payments. With the refactor, the need for that has been eliminated. 

More specifically, instead of passing around two map objects describing gateways, one for installed, and one for suggestions, we're just passing around one map for suggested gateways and enriching it with any required properties from the Payment Gateway API.

### Detailed test instructions:

-   Enable remote payments
-   Use default gateways (no remote payment tester required)
- General smoke test, ensuring that plugin installs are working as expected
- Disable remote payments (use local)
- Set store to USA
- Ensure that WCPay card is displayed, and works properly when you hit "Get Started"
